Gems/BarrierInput/Code/Source/BarrierInputMouse.cpp

```diff
@@ -0,0 +1,194 @@
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    InputDeviceMouseBarrier::InputDeviceMouseBarrier(InputDeviceMouse& inputDevice)
+        : InputDeviceMouse::Implementation(inputDevice)
+        , m_systemCursorState(SystemCursorState::Unknown)
+        , m_systemCursorPositionNormalized(0.5f, 0.5f)
+        , m_threadAwareRawButtonEventQueuesById()
+        , m_threadAwareRawButtonEventQueuesByIdMutex()
+        , m_threadAwareRawMovementEventQueuesById()
+        , m_threadAwareRawMovementEventQueuesByIdMutex()
+        , m_threadAwareSystemCursorPosition(0.0f, 0.0f)
+        , m_threadAwareSystemCursorPositionMutex()
+    {
+        RawInputNotificationBusBarrier::Handler::BusConnect();
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    InputDeviceMouseBarrier::~InputDeviceMouseBarrier()
+    {
+        RawInputNotificationBusBarrier::Handler::BusDisconnect();
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    bool InputDeviceMouseBarrier::IsConnected() const
+    {
+        // We could check the validity of the socket connection to the Barrier server
+        return true;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceMouseBarrier::SetSystemCursorState(SystemCursorState systemCursorState)
+    {
+        // This doesn't apply when using Barrier, but we'll store it so it can be queried
+        m_systemCursorState = systemCursorState;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    SystemCursorState InputDeviceMouseBarrier::GetSystemCursorState() const
+    {
+        return m_systemCursorState;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceMouseBarrier::SetSystemCursorPositionNormalized(AZ::Vector2 positionNormalized)
+    {
+        // This will simply get overridden by the next call to OnRawMousePositionEvent, but there's
+        // not much we can do about it, and Barrier mouse input is only for debug purposes anyway.
+        m_systemCursorPositionNormalized = positionNormalized;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    AZ::Vector2 InputDeviceMouseBarrier::GetSystemCursorPositionNormalized() const
+    {
+        return m_systemCursorPositionNormalized;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceMouseBarrier::TickInputDevice()
+    {
+        {
+            // Queue all mouse button events that were received in the other thread
+            AZStd::lock_guard<AZStd::mutex> lock(m_threadAwareRawButtonEventQueuesByIdMutex);
```

Comment:
FYI you don't need to specify `AZStd::mutex` due to [Class Template Argument Deduction](https://en.cppreference.com/w/cpp/language/class_template_argument_deduction)
```suggestion
            AZStd::scoped_lock lock(m_threadAwareRawButtonEventQueuesByIdMutex);
```

Reply:
Updated all instances of this in BarrierInputMouse.cpp and BarrierInputKeyboard.cpp